demos/python_demos/face_recognition_demo/face_recognition_demo.py

```diff
@@ -0,0 +1,1071 @@
+            rois = rois[:self.QUEUE_SIZE]
+        self.head_pose_estimator.start_async(frame, rois)
+        self.landmarks_detector.start_async(frame, rois)
+        head_poses = self.head_pose_estimator.get_head_poses()
+        landmarks = self.landmarks_detector.get_landmarks()
+
+        self.face_identifier.start_async(frame, rois, landmarks)
+        face_identities = self.face_identifier.get_matches()
+
+        outputs = [rois, head_poses, landmarks, face_identities]
+
+        return outputs
+
+
+    def get_performance_stats(self):
+        stats = {
+            'face_detector': self.face_detector.get_performance_stats(),
+            'head_pose': self.head_pose_estimator.get_performance_stats(),
+            'landmarks': self.landmarks_detector.get_performance_stats(),
+            'face_identifier': self.face_identifier.get_performance_stats(),
+        }
+        return stats
+
+
+class Visualizer:
+    DEFAULT_CAMERA_FOCAL_DISTANCE = 950.0
+
+    BREAK_KEY_LABEL = 'q'
+    BREAK_KEY = ord(BREAK_KEY_LABEL)
+
+
+    def __init__(self, args):
+        self.frame_processor = FrameProcessor(args)
+        self.display = not args.no_show
+        self.print_perf_stats = args.perf_stats
+
+        self.frame_time = 0
+        self.frame_start_time = 0
+        self.fps = 0
+
+        self.input_crop = None
+        if args.crop_width and args.crop_height:
+            self.input_crop = np.array((args.crop_width, args.crop_height))
+
+        self.frame_timeout = 0 if args.timelapse else 1
+
+    def update_fps(self):
+        now = time.time()
+        self.frame_time = now - self.frame_start_time
+        self.fps = 1.0 / self.frame_time
+        self.frame_start_time = now
+
+    def draw_text_with_background(self, frame, text, origin,
+            font=cv2.FONT_HERSHEY_SIMPLEX, scale=1.0,
+            color=(0, 0, 0), thickness=1, bgcolor=(255, 255, 255)):
+        text_size, baseline = cv2.getTextSize(text, font, scale, thickness)
+        cv2.rectangle(frame,
+            tuple((origin + (0, baseline)).astype(int)),
+            tuple((origin + (text_size[0], -text_size[1])).astype(int)),
+            bgcolor, cv2.FILLED)
+        cv2.putText(frame, text,
+            tuple(origin.astype(int)),
+            font, scale, color, thickness)
+        return text_size, baseline
+
+    def draw_detection_roi(self, frame, roi, identity):
+        label = self.frame_processor \
+            .face_identifier.get_identity_label(identity.id)
+
+        # Draw face ROI border
+        cv2.rectangle(frame,
+            tuple(roi.position), tuple(roi.position + roi.size),
+            (0, 220, 0), 2)
+
+        # Draw identity label
+        text_scale = 0.5
+        font = cv2.FONT_HERSHEY_SIMPLEX
+        text_size = cv2.getTextSize("H1", font, text_scale, 1)
+        line_height = np.array([0, text_size[0][1]])
+        self.draw_text_with_background(frame,
+            '%s %.2f%%' % (label, 100.0 * (1 - identity.distance)),
```

Comment:
In case of *Unknown* class, expression `100.0 * (1 - identity.distance)` no more represents confidence degree, so it's better not to show it to the user.

Reply:
The goal was to display model confidences to the user so that he could adjust the threshold (`-t_id`) used for person identification. Thus, this feature is mostly for debugging. May be it should be displayed only in `--verbose` mode?